File: CarePackage/Software/Postman.cs
namespace CarePackage.Software;

public class Postman : BaseSoftware
{
    public override string           Key            { get; set; } = "postman";
    public override string           Name           { get; set; } = "Postman";
    public override SoftwareCategory Category       { get; set; } = SoftwareCategory.Development;
    public override string           DownloadName   { get; set; } = "Postman-win64-Setup.exe";
    public override bool             IsArchive      { get; set; } = false;
    public override bool             ShouldCacheUrl { get; set; } = false;
    public override bool             RequiresAdmin  { get; set; } = false;
    public override Bitmap           Icon           { get; set; } = Resources.Icons.postman;
    public override string           Homepage       { get; set; } = "https://postman.com";

    public override Task<string> GetDownloadUrlAsync(CancellationToken ct)
        => Task.FromResult("https://dl.pstmn.io/download/latest/win64");
}

File: packages/app/src/services/software/definitions/postman.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'postman';
	public name = 'Postman';
	public category = [SoftwareCategory.Development, SoftwareCategory.Network];
	public downloadName = 'Postman-win64-Setup.exe';
	public icon = 'postman.png';
	public homepage = 'https://postman.com';

	public async resolveDownloadUrl() {
		return ok('https://dl.pstmn.io/download/latest/win64');
	}
}
